ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/events/AlterTableEvent.java

```diff
@@ -101,6 +112,36 @@ private HiveOperationType getOperationType() {
       ret.add(getHivePrivilegeObjectDfsUri(newUri));
     }
 
+    if(newTable.getParameters().containsKey(hive_metastoreConstants.META_TABLE_STORAGE)) {
+      String storageUri = "";
+      DefaultStorageHandler defaultStorageHandler = null;
+      HiveStorageHandler hiveStorageHandler = null;
+      Configuration conf = new Configuration();
+      Map<String, String> tableProperties = new HashMap<>();
+      tableProperties.putAll(newTable.getSd().getSerdeInfo().getParameters());
+      tableProperties.putAll(newTable.getParameters());
+      try {
+        hiveStorageHandler = (HiveStorageHandler) ReflectionUtils.newInstance(
+                conf.getClassByName(newTable.getParameters().get(hive_metastoreConstants.META_TABLE_STORAGE)), event.getHandler().getConf());
+        Method methodIsImplemented = hiveStorageHandler.getClass().getMethod("getURIForAuth", Map.class);
+        if(methodIsImplemented != null && hiveStorageHandler instanceof DefaultStorageHandler) {
```

Comment:
isnt it enough to just check if the storagehandler class implements the new interface?
something like this
(hiveStorageHandler instanceof HiveStorageAuthorizationHandler)

So this would be true for all the Hive implemented ones and custom handlers that dont implement this wouldnt use this URI Authorization right ?